spec/request/api/registration/registrations_spec.rb

```diff
@@ -0,0 +1,122 @@
+             role: "client",
+             company_name: "company name",
+             company_url: "http://develup.com",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message" do
+      expect(response_json["errors"]["full_messages"][0]).to eq "Password confirmation doesn't match Password"
+    end
+  end
+
+  describe "Unsuccessfully register - with invalid email" do
+    before do
+      post "/api/auth",
+           params: {
+             email: "clien tttttt@mail",
+             password: "password",
+             password_confirmation: "password",
+             role: "client",
+             company_name: "company name",
+             company_url: "http://develup.com",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message" do
+      expect(response_json["errors"]["email"][0]).to eq "is not an email"
+    end
+  end
+
+  describe "Unsuccesfuly register - email already exists" do
+    let!(:user) { create(:user, email: "client@mail.com") }
+    before do
+      post "/api/auth",
+           params: {
+             email: "client@mail.com",
+             password: "password",
+             password_confirmation: "password",
+             role: "client",
+             company_name: "company name",
+             company_url: "http://develup.com",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message" do
+      expect(response_json["errors"]["full_messages"][0]).to eq "Email has already been taken"
+    end
+  end
+
+  describe "Unsuccesfuly register -with blank company_name, company_url and role" do
+    before do
+      post "/api/auth",
+           params: {
+             email: "client@mail.com",
+             password: "password",
+             password_confirmation: "password",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message if role is blank" do
+      expect(response_json["errors"]["full_messages"][0]).to eq "Role can't be blank"
+    end
+    it "returns a unsuccesfully message if company name is blank" do
+      expect(response_json["errors"]["full_messages"][1]).to eq "Company name can't be blank"
+    end
+    it "returns a unsuccesfully message if company url is blank" do
+      expect(response_json["errors"]["full_messages"][2]).to eq "Company url can't be blank"
```

Comment:
is expected

Reply:
I have modified the tests to differentiate unit from acceptance.